src/main.ts

```diff
@@ -0,0 +1,32 @@
+import * as core from "@actions/core";
+import { loadSecrets, unsetPrevious, validateAuth } from "./utils";
+
+export const run = async (): Promise<void> => {
+	try {
+		// Get action inputs
+		const shouldUnsetPrevious = core.getBooleanInput("unset-previous");
+		const shouldExportEnv = core.getBooleanInput("export-env");
+
+		// Unset all secrets managed by 1Password if `unset-previous` is set.
+		if (shouldUnsetPrevious) {
+			unsetPrevious();
+		}
+
+		// Validate that a proper authentication configuration is set for the CLI
+		validateAuth();
```

Comment:
I've noticed that we validate the auth here, but later we check for environment variables a couple of more times:
- when choosing the type of client to use for resolving secret references.
- when building each of the clients

The auth validation was done before everything else before to avoid installing the CLI if the authentication wasn't configured as expected and stop the action early. Since now this step is removed, we can simplify this by doing one of the following to simplify things a bit:
1. Tweak the `validateAuth` to return a type with the values already extracted from the environment variables. Then in `loadSecrets` instead of doing `buildSecretResolver` you can directly build the resolver by doing something like this:
   ```ts
   let auth = validateAuth();

   // ...

   const resolver =
		"host" in auth ? new Connect(auth.host, auth.token) : new Account(auth.token);
   ```

2. We can simply merge the functionality of `validateAuth` and `buildSecretResolver` into a function (e.g. named `getAuth`) which will return the resolver. It could look like this:
   ```ts
   const getAuth = (): SecretReferenceResolver => {
       // same code as in validateAuth

       if (authType === "Connect") {
           return new Connect(process.env[envConnectHost], process.env[envConnectToken]);
       } else {
           return new Account(process.env(envServiceAccountToken];
       }
   }
   ```

Note that in both cases you'd need to tweak the constructors for `Connect` and `Account` classes to accept these parameters.

You can choose the option you feel fits best here. 😄

Reply:
implment in 087ae1055654ea85f3a4ebddc98c25117529c669 test in afa821d54035fc9e6167d801ebf5dfb9bc384acd

Reply:
The latest implementation of this is so elegant. Thank you for addressing this feedback. 😄